apps/docs/guide/deploying/validator.md

```diff
@@ -10,50 +10,141 @@ The following diagram illustrates the deployment system implemented by Sunodo.
+
+-   [Alchemy](http://alchemy.com)
+-   [Infura](https://infura.io)
+-   [Chainstack](http://chainstack.com)
+-   [GetBlock](https://getblock.io)
 
-::: info
-Add more information about wallet setup
+### Database
+
+The node uses a [postgres](http://postgresql.org) database to store application state (inputs and outputs), and serve those over a GraphQL API. The user must [setup a database server](https://www.postgresql.org/download/), or use a third-party managed database provider.
+
+The database credentials provided to the Sunodo node must have permissions to create new databases, because each application will have its own independent database. The schema of the application database does not need to be created beforehand because it's dynamically created by the Sunodo node.
+
+The `sunodo-validator start` command provides the option `--embedded-database` that will spin up a postgres instance. This option must only be used for test purposes, because persistence of data is turned off, there is no backup mechanism, and the security of the database access is not production-grade.
+
+### Wallet
+
+Sunodo validator node interacts with the `Authority` smart contract created above to submit claims of the application state once every epoch. In order to do that it needs to be configured with the base layer wallet that is the owner of the `Authority` contract.
+
+::: warning
+A wallet must be used by a single Sunodo node. Do not share the wallet with other applications or other Sunodo nodes as the nonce management may cause issues.
 :::
 
-## Setting up the Controller
+The first and less safe way is to specify a mnemonic passphrase as the `--mnemonic-passphrase` command option and optionally the `--mnemonic-index` which defaults to zero. The environment variable `CARTESI_AUTH_MNEMONIC` can also be used instead.
+
+A much safer way to specify the wallet is using [AWS KMS](https://docs.aws.amazon.com/kms/) as the private key manager. This is the recommended method for production applications, even if the node is not running inside AWS.
 
-The Sunodo controller's job is to watch an specific `ValidatorNodeProvider` contract and add or remove applications from a Sunodo node based on funding information.
+Refer to [this blog post](https://aws.amazon.com/blogs/database/part1-use-aws-kms-to-securely-manage-ethereum-accounts/) that explains in details how to create a AWS KMS key for Ethereum accounts. Once you create a key you can specify the KMS key id using the `--kms-key-id` command line option and provide the appropriate AWS credentials that have access to use the key.
 
-::: info
-Add more information about the controller
+:::info
+A AWS KMS key is much safer because the private key itself is not exposed to the Sunodo node application. What is exposed is the access to **use** the key, which can be revoked by a master AWS account. On the other hand compromising the private key itself has no turning point.
```

Comment:
```suggestion
An AWS KMS key is much safer because the private key itself is not exposed to the Sunodo node application. What is exposed is the access to **use** the key, which can be revoked by a master AWS account. On the other hand, compromising the private key itself has no turning point.
```